openpype/tools/attribute_defs/widgets.py

```diff
@@ -442,17 +454,40 @@ def _on_value_change(self):
         self.value_changed.emit(new_value, self.attr_def.id)
 
     def current_value(self):
+        if self.multiselection:
+            return self._input_widget.value()
         idx = self._input_widget.currentIndex()
         return self._input_widget.itemData(idx)
 
+    def _multiselection_multivalue_prep(self, values):
+        final = None
+        multivalue = False
+        for value in values:
+            value = set(value)
+            if final is None:
+                final = value
+            elif multivalue or final != value:
+                final |= value
+                multivalue = True
```

Comment:
```suggestion
            if final is None:
                final = value
            elif multivalue and not final:
                break
            elif multivalue or final != value:
                final |= value
                multivalue = True
```

Reply:
I'm doing Union, not intersection.

Reply:
Damn those operator overloads - thanks. I clearly need to just go to bed tonight.